### PR TITLE
Bob/7436 syphilis card support pt 5

### LINKED ISSUE #7436: [Syphilis] Test card supports Syphilis test results
## Why should we do this?
- (current experience) The test card screen currently does not support syphilis-specific options.
- (why we want to make this change) When a user selects a syphilis device on the test card screen in SimpleReport, they should see options for syphilis test results and see the appropriate AOE questions.

## User story
As a SimpleReport user, I want to be able to select the appropriate syphilis test result option on the test card screen when using a syphilis device so that I can accurately record and manage syphilis test results for my patients, ensuring the correct interpretation and documentation of their syphilis status.

## Action Requested
- Syphilis test result options (e.g., negative, positive, inconclusive) are displayed when a syphilis device is selected on the test card screen.
- The syphilis test result values are mapped correctly to the corresponding LOINC codes.
- Syphilis is added as a supported disease in our database.
- Syphilis specific AOE questions are displayed on the test card
- Update FHIR converter logic to include syphilis results and AOE responses in FHIR bundle

## Acceptance Criteria

**Review**
-   [ ] Reviewed with design and/or design lead
-   [ ] Reviewed with content strategist for final approval
-   [ ] Reviewed with product and engineering

## Additional Context
**Syphilis AOE Questions:** The following information is to be presented on the screen for single-entry when the patient/participant has a positive/reactive Syphilis test result. _(text below is meant as a placeholder until final confirmation is received from DHSP)_

```
Syphilis result
[Radio button] Positive
[Radio button] Negative
[Radio button] Inconclusive

Has the patient been told they have syphilis before?
[Radio button] Yes
[Radio button] No
[Radio button] Prefer not to say

Is the patient currently experiencing any symptoms?
[Radio button] Yes
[Radio button] No

Which symptoms are they experiencing?
[checkbox] Genital sore/lesion
[checkbox] Anal sore/lesion
[checkbox] Sore(s) in mouth/lips
[checkbox] Body Rash
[checkbox] Palmar (hand)/plantar (foot) rash
[checkbox] Flat white warts
[checkbox] Hearing loss
[checkbox] Blurred vision
[checkbox] Patchy hair loss

Date of symptom onset
MM/DD/YYYY

What is the gender of their sexual partners? (Select all that apply)
[checkbox] Female
[checkbox] Male
[checkbox] Transwoman
[checkbox] Transman
[checkbox] Nonbinary or gender non-conforming
[checkbox] Gender identity not listed here
[checkbox] Prefer not to answer

Is the patient pregnant?
[Radio button] Yes
[Radio button] No
[Radio button] Prefer not to say

[Button] Submit results
```

### PR DESCRIPTION
**Pull request by fzhao99**
# FRONTEND PULL REQUEST

## Related Issue
- Resolves #7436 
- Part 5 of the syphilis work that finishes on the previous PR's
     - See [part 4](https://github.com/CDCgov/prime-simplereport/pull/7713) for more context
- PR size is mostly a snapshot generation: actual SOLC is 300 lines or so

## Changes Proposed
- Calls the logic put in in part 4 to show the syphilis AOE form when the right device is selected
- Builds out the AOE form for syphilis
     - Uses the [listener generation hooks](https://github.com/CDCgov/prime-simplereport/blob/264a8ccd00f5fb1f707f8b49d63c91cbc302bf59/frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/aoeUtils.ts#L17) put in during part 4 
- Displays syphilis symptoms in a column format as defined by design
- Writes some relevant tests
     - The mocking setup is kinda terrible because our Apollo setup requires us to extend a mock each time there's a card interaction. Upshot is that we have mocks with super long names that iteratively include data related to each test click, and I decided to err on the side of verbosity / tried to constrain the grossness just to the individual mock file. Briefly toyed around with some sort of factory pattern to try and generate mocks but aborted since don't think these mocks are going to have super high churn, but if anyone has any suggestions here about what to do that'd be great. 

## Testing
- Validate the syphilis interactions are working as expected. Deployed on dev 3
https://github.com/CDCgov/prime-simplereport/assets/29645040/2bce9171-f669-4731-855a-9a448d5ed58a

### CODE REVIEW

File: frontend/src/app/testQueue/TestCard/TestCard.test.tsx [hunk 1]
```diff
@@ -1117,33 +1168,244 @@ describe("TestCard", () => {
       ).not.toBeInTheDocument();
     });
 
-    it("should show no AOE questions when a flu only device is chosen", async function () {
```
Comment (fzhao99) [inline frontend/src/app/testQueue/TestCard/TestCard.test.tsx:1120]: Moved this one up to be next to other flu-only tests

**Review by mpbrown — APPROVED** [review]
LGTM on dev3! It does look like we'll need to update the results page to handle the new symptoms AOE data since it doesn't show for the details modal, I think it makes sense to resolve it in a separate ticket though

image

File: frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/SyphilisAoEForm.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+          </div>
+
+          <TextInput
+            data-testid="symptom-date"
+            name={`symptom-date-${testOrder.internalId}`}
+            type="date"
+            required
+            label="Date of symptom onset"
+            aria-label="Symptom onset date"
+            min={formatDate(new Date("Jan 1, 2020"))}
```
Comment (emyl3) [inline frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/SyphilisAoEForm.tsx:169]: should this be more flexible? I actually have no idea 🤔

**Review by emyl3 — APPROVED** [review]
Left one question -- LGTM!

File: frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/SyphilisAoEForm.tsx (hunk 2 above)
Comment (fzhao99) [reply frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/SyphilisAoEForm.tsx:169]: I think we do something similar for the COVID one (https://github.com/CDCgov/prime-simplereport/blob/main/frontend/src/app/testQueue/TestCardForm/diseaseSpecificComponents/CovidAoEForm.tsx) and I think functionally no one's going to backdate before the beginning of the pandemic. Think it's fine to leave it as is

**Review by bobbywells52 — APPROVED** [review]
Tested in dev3 and is working for me -- LGTM! Great work with the tests and mocks too! I can't believe how fast you are flying through this 🤯

**Pull request merged**